UnitystationLauncher/Services/ServerAuthenticationService.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Security.Authentication;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Newtonsoft.Json;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Digests;
+using Org.BouncyCastle.Crypto.Encodings;
+using Org.BouncyCastle.Crypto.Engines;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+using Org.BouncyCastle.OpenSsl;
+using UnitystationLauncher.Models;
+using UnitystationLauncher.Models.Api;
+
+namespace UnitystationLauncher.Services;
+
+public interface IServerAuthenticationService
+{
+    public Task<Server> GetServerInfoByIP(string IP);
+
+    public Task<Dictionary<string, string>> AuthenticateWithServer(string IP, Installation Installation);
+}
+
+public class ServerAuthenticationService : IServerAuthenticationService
+{
+    public ServerAuthenticationService(AuthService AuthService)
+    {
+        _AuthService = AuthService;
+    }
+
+
+    private readonly AuthService _AuthService;
+    private readonly HttpClient _httpClient = new HttpClient();
+
+    private readonly SHA512 SHA512 = SHA512.Create();
+
+    public async Task<Server> GetServerInfoByIP(string IP)
+    {
+        try
+        {
+            string Port = "7778";
```

Comment:
hard coded port :(
i will look into if its possible to share the same port between mirror and the query server